src/Language/PureScript/Pretty/Common.hs

```diff
@@ -26,9 +32,87 @@ import Text.PrettyPrint.Boxes
 -- Wrap a string in parentheses
 --
 parens :: String -> String
-parens s = ('(':s) ++ ")"
+parens s = '(':s ++ ")"
+
+parensPos :: (Emit gen) => gen -> gen
+parensPos s = emit "(" `mappend` s `mappend` emit ")"
+
+-- |
+-- Generalize intercalate slightly for monoids
+--
+intercalate :: Monoid m => m -> [m] -> m
+intercalate x xs = mconcat (intersperse x xs)
+
+class (Monoid gen) => Emit gen where
+  emit :: String -> gen
+  addMapping :: (Maybe SourceSpan) -> gen
```

Comment:
Does it make sense to simplify this to `addMapping :: SourceSpan -> gen` and to send `Nothing` to `mempty` in a helper function instead?